src/commands/dao_cmds/apps.js

```diff
@@ -69,34 +75,73 @@ exports.handler = async function({
           })
         },
       },
+      {
+        title: 'Fetching permissionless apps',
+        enabled: () => all,
+        task: async (ctx, task) => {
+          const kernel = new web3.eth.Contract(
+            getContract('@aragon/os', 'Kernel').abi,
+            ctx.daoAddress
+          )
+
+          const events = await kernel.getPastEvents('NewAppProxy', {
+            fromBlock: 0,
+            toBlock: 'latest',
+          })
+
+          ctx.appsWithoutPermissions = events
+            .map(event => ({
+              proxyAddress: event.returnValues.proxy,
+              appId: event.returnValues.appId,
+            }))
+            // Remove apps that have permissions
+            .filter(
+              ({ proxyAddress }) =>
+                !ctx.apps.find(app =>
+                  addressesEqual(app.proxyAddress, proxyAddress)
+                )
+            )
+        },
+      },
     ],
     listrOpts(silent, debug)
   )
 
   return tasks.run().then(ctx => {
     reporter.success(`Successfully fetched DAO apps for ${ctx.daoAddress}`)
-    const appsContent = ctx.apps.map(
-      ({ appId, proxyAddress, codeAddress, content, appName, version }) => [
-        appName ? `${appName}@v${version}` : printAppName(appId),
-        proxyAddress,
-        printContent(content),
-      ]
-    )
-
-    // filter registry name to make it shorter
-    // TODO: Add flag to turn off
-    const filteredContent = appsContent.map(row => {
-      row[0] = row[0].replace('.aragonpm.eth', '')
-      return row
-    })
+    const appsContent = ctx.apps
+      .map(
+        ({ appId, proxyAddress, codeAddress, content, appName, version }) => [
+          appName ? `${appName}@v${version}` : printAppName(appId),
+          proxyAddress,
+          printContent(content),
+        ]
+      )
+      // filter registry name to make it shorter
+      // TODO: Add flag to turn off
+      .map(row => {
+        row[0] = row[0].replace('.aragonpm.eth', '')
+        return row
+      })
 
     const table = new Table({
       head: ['App', 'Proxy address', 'Content'].map(x => chalk.white(x)),
     })
+    appsContent.forEach(row => table.push(row))
+    console.log(table.toString())
 
-    filteredContent.forEach(row => table.push(row))
+    // Print permisionless apps
+    const table2 = new Table({
```

Comment:
Could probably come up with better names for these table variables 😄

Reply:
See all evil? :smile: